### PR TITLE
Chapter 6- Modelling User

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**
Chapter 6 Modelling User

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    length: { maximum: Settings.validations.email.max_length },
+    format: { with: Settings.validations.email.regex },
+    uniqueness: { case_sensitive: true }
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
+
+  has_secure_password
+  
+  before_save { email.downcase! }
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  before_save downcase_email
  
  private 
  
  def downcase_email
    email.downcase!
  end
```

**Comment by EvonneNgXinYi** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class User < ApplicationRecord
+
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
```

**Review by saobv-2223 — COMMENTED** [review]
fix small mistake and merge

**Pull request merged**